src/odemis/util/synthetic.py

```
@@ -125,3 +126,493 @@ def psf_gaussian(
+        :return: 2D array representing the simulated intensity pattern on the camera.
+        """
+        if self._last_pos != current_pos:
+            try:
+                logging.debug("Simulating new ray-traced image")
+                dx = self._aligned_pos["x"] - current_pos["x"]
+                dy = self._aligned_pos["y"] - current_pos["y"]
+                dz = self._aligned_pos["z"] - current_pos["z"]
+                with numpy.errstate(all="raise"):
+                    self._last_img = self._get_ray_traced_pattern(dx, dy, dz)
+                self._last_pos = current_pos.copy()
+            except Exception:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)